### PR TITLE
RunningInKuberenes flag (used in API handler registration)

### LINKED ISSUE #575: Do not export Kube specific model endpoints in non-kube clusters
@afein: Can we avoid exposing endpoints specific to kubernetes clusters like `namespaces` and `pods` from non-kubernetes clusters? AFAIK, the other endpoints like containers, nodes and cluster should still continue to work.

### PR DESCRIPTION
**Pull request by mwielgus**
Fixes #575

### CODE REVIEW

File: heapster.go [hunk 1]
```diff
@@ -34,16 +34,17 @@ import (
...
-	argModelResolution = flag.Duration("model_resolution", 1*time.Minute, "The resolution of the timeseries stored in the model. Applies only if use_model is true")
-	argPort            = flag.Int("port", 8082, "port to listen to")
-	argIp              = flag.String("listen_ip", "", "IP to listen on, defaults to all IPs")
-	argMaxProcs        = flag.Int("max_procs", 0, "max number of CPUs that can be used simultaneously. Less than 1 for default (number of cores)")
-	argSources         manager.Uris
-	argSinks           manager.Uris
+	argStatsResolution     = flag.Duration("stats_resolution", 5*time.Second, "The resolution at which heapster will retain stats.")
+	argSinkFrequency       = flag.Duration("sink_frequency", 10*time.Second, "Frequency at which data will be pushed to sinks")
+	argCacheDuration       = flag.Duration("cache_duration", 4*time.Minute, "The total duration of the historical data that will be cached by heapster.")
+	argRunningInKuberentes = flag.Bool("running_in_kubernetes", true, "When true, the kubernetes-specific api will be enabled")
```
Comment (vishh) [inline heapster.go]: Can we somehow autodetect this? For example, if one of the sources is kubernetes, we can use that as an hint. Default values for flags like this affect usability. WDYT?

**Comment by vishh** [comment]
@mwielgus: Thank you for sending out this PR :)
Just one comment. Otherwise LGTM.

File: heapster.go (hunk 1 above)
Comment (mwielgus) [inline heapster.go]: Added detection.

**Comment by vishh** [comment]
Awesome! LGTM!

**Pull request merged**